cdap-app-fabric/src/main/java/co/cask/cdap/metadata/CollapsedRelation.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.metadata;
+
+import co.cask.cdap.data2.metadata.lineage.AccessType;
+import co.cask.cdap.data2.metadata.lineage.Relation;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.metadata.lineage.CollapseType;
+import com.google.common.collect.ImmutableSet;
+import org.apache.twill.api.RunId;
+
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A {@link Relation} collapsed based on {@link CollapseType}.
+ */
+public class CollapsedRelation {
+  private final Id.NamespacedId data;
```

Comment:
Unfortunately, using newer Id classes leads to lot of glue code that will again have to be cleaned up later. Instead we can just do one pass later to use newer Id classes.